lib/ts/hugepages.h

```diff
@@ -20,13 +20,26 @@
  */
 #ifndef _hugepages_h_
 #define _hugepages_h_
+#include "ink_memory.h"
 
 #include <cstring>
 
+#define HUGEPAGE_SIZE  (PAGE_SIZE<<9)
+
 size_t ats_hugepage_size(void);
 bool ats_hugepage_enabled(void);
 void ats_hugepage_init(int);
 void *ats_alloc_hugepage(size_t);
 bool ats_free_hugepage(void *, size_t);
+void *ats_alloc_hugepage_stack(size_t);
+
+void *huge_normal_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+void *huge_nodump_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+bool huge_dalloc(void *chunk, size_t size, bool committed, unsigned arena_ind);
+bool huge_commit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_decommit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_purge(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_split(void *chunk, size_t size, size_t size_a, size_t size_b, bool committed, unsigned arena_ind);
+bool huge_merge(void *chunk_a, size_t size_a, void *chunk_b, size_t size_b, bool committed, unsigned arena_ind);
```

Comment:
Main idea of why we are using these shims at all is (1) Jemalloc doesn't support aligned hugepage-prepared pages as not everyone demands them as much as we and (2) Jemalloc allows shims specifically per-Arena therefore we can pre-MADV_NODUMP pages and segments instead of per-malloc() [ewww].

So ... in these two cases we treat pages in a custom way from the free pages.   We turn a flag on in one case and prepare large blocks from smaller requests in the other case [more common for us].

There's no other place in the Jemalloc system to handle pages/mapping segments when requested other than these hooks.

I should repeat: this is stateless. This code doesn't "handle" segments or even remember which are which.   It just prepares them and operates on them later when Jemalloc has various needs.